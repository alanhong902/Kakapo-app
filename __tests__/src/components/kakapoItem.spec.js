import React from 'react';
import { map } from 'ramda';
import { newSoundObj } from 'utils/';
import { ImportKakapoItem as KakapoItem } from 'components/';
import { getData, createComponentWithIntl } from '../helper';

const soundProp = (props = {}) => {
  const obj = { ...newSoundObj, source: 'file', progress: 1, ...props };
  return { sound: map(x => (x === null ? 'wind' : x), obj) };
};

function setup(props = {}) {
  const propData = {
    ...getData('sounds', { full: true }),
    ...props,
  };
  return {
    props,
    tree: createComponentWithIntl(<KakapoItem {...propData} />).toJSON(),
  };
}

test('<KakapoItem/> render', () => {
  const { tree } = setup(soundProp());
  expect(tree).toMatchSnapshot();
});

test('<KakapoItem/> render file', () => {
  const { tree } = setup(soundProp({ file: 'someNewFile' }));
  expect(tree).toMatchSnapshot();
});
