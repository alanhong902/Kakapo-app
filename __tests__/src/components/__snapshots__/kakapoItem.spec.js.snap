// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<KakapoItem/> render 1`] = `
<div
  className="kakapo-item disabled"
  onClick={[Function]}
  role="link"
  tabIndex={-1}
>
  <div
    className="thumbnail"
  >
    <i
      className="icon-wind"
    />
  </div>
  <span
    className="title"
  >
    <span>
      Wind
    </span>
  </span>
</div>
`;

exports[`<KakapoItem/> render file 1`] = `
<div
  className="kakapo-item"
  onClick={[Function]}
  role="link"
  tabIndex={-1}
>
  <div
    className="thumbnail"
  >
    <i
      className="icon-wind"
    />
  </div>
  <span
    className="title"
  >
    <span>
      Wind
    </span>
  </span>
</div>
`;
